File: typescript/examples/getting-started/13-web-tools.ts
// Web tools: search the web and fetch URL content
import { Opper } from "../../src/index.js";

const opper = new Opper();

// ── Web search ──────────────────────────────────────────────────────────────

console.log("── Web search ──");
const searchResults = await opper.beta.web.search({ query: "Opper AI platform" });

for (const result of searchResults.results.slice(0, 5)) {
  console.log(`  ${result.title}`);
  console.log(`  ${result.url}`);
  console.log(`  ${result.snippet}\n`);
}

// ── Fetch URL content ───────────────────────────────────────────────────────

console.log("── Fetch URL ──");
const page = await opper.beta.web.fetch({ url: "https://docs.opper.ai" });
console.log("Content (first 500 chars):", page.content.slice(0, 500) + "...");

File: python/examples/getting-started/13_web_tools.py
# Web tools: search the web and fetch URL content
from opperai import Opper

opper = Opper()

# ── Web search ──────────────────────────────────────────────────────────────

print("── Web search ──")
search_results = opper.beta.web.search(query="Opper AI platform")

for result in search_results.results[:5]:
    print(f"  {result.title}")
    print(f"  {result.url}")
    print(f"  {result.snippet}\n")

# ── Fetch URL content ───────────────────────────────────────────────────────

print("── Fetch URL ──")
page = opper.beta.web.fetch(url="https://docs.opper.ai")
print("Content (first 500 chars):", page.content[:500] + "...")
